add_organism.py

```diff
@@ -0,0 +1,141 @@
+
+    crisprfinder_script = subprocess.Popen(['perl' ,'CRISPRCasFinder/CRISPRCasFinder.pl','-out', CRISPRFinder_output_folder, '-in', 'seq.fasta'], stdout=sys.stdout)
+    crisprfinder_script.communicate()
+
+    results = json.load(open(CRISPRFinder_output_folder + '/result.json'))
+
+    for seq in results['Sequences']:
+        for cspr in seq['Crisprs']:
+            loc_start = int(cspr['Start'])
+            loc_end = int(cspr['End'])
+            dr_consensus = cspr['DR_Consensus']
+            orientation = cspr['Potential_Orientation'] == '+'
+            locus, _ = Locus.objects.get_or_create(
+                organism=o,
+                genomic_start=loc_start,
+                genomic_end=loc_end,
+                consensus=dr_consensus,
+                orientation=orientation
+            )
+            for i in range(1, len(cspr['Regions'])-2, 2):
+                rep = cspr['Regions'][i]
+                spac = cspr['Regions'][i+1]
+                if rep['Type'] != 'DR':
+                    print('First region in repeat/spacer pair is not a repeat!')
+                    print('Organism: ', o)
+                    return
+                elif spac['Type'] != 'Spacer':
+                    print('Second region in repeat/spacer pair is not a spacer!')
+                    print('Organism: ', o)
+                    return
+
+                order = (i-1)/2 + 1 #starting at 1
+                repeat, _ = Repeat.objects.get_or_create(sequence=rep['Sequence'])
+                spacer, _ = Spacer.objects.get_or_create(sequence=spac['Sequence'])
+
+                lsr, _ = LocusSpacerRepeat.objects.get_or_create(
+                    locus=locus,
+                    spacer=spacer,
+                    repeat=repeat,
+                    order=order
+                )
+            rep = cspr['Regions'][-2]
+            if rep['Type'] != 'DR':
+                print('Last region in locus is not a repeat!')
+                print('Organism: ', o)
+                return
+            order = (len(cspr['Regions'])-3)/2 + 1
+            repeat, _ = Repeat.objects.get_or_create(sequence=rep['Sequence'])
+            lsr, _ = LocusSpacerRepeat.objects.get_or_create(
+                locus=locus,
+                spacer=None,
+                repeat=repeat,
+                order=order
+            )
+
+def get_ncbi_seq(email, db, rettype, accession):
+
+    # fetch
+    print("Fetching accession %s from GenBank\n" % (accession))
+
+    Entrez.email = email
+
+    try:
+        handle = Entrez.efetch(
+            db=db,
+            rettype=rettype,
+            retmode="text",
+            id=accession
+        )
+        res = handle.read()
+
+        # for testing only
+        pickle_item(res, accession)
+        return res
+
+    except Exception:
+        sys.stderr.write("Error! Cannot fetch: %s        \n" % accession)
+
+if __name__ == '__main__':
+    add_sequence_to_db(accession='NC_000853', name='Thermotoga maritima MSB8')
```

Comment:
Also make the accession a command-line argument.

Reply:
Fixed :)